plugins/org.locationtech.udig.project.tests/src/org/locationtech/udig/project/internal/render/impl/renderercreator/SingleFeatureRenderMetricsFactory2.java

```diff
@@ -0,0 +1,75 @@
+/* uDig - User Friendly Desktop Internet GIS client
+ * http://udig.refractions.net
+ * (C) 2004, Refractions Research Inc.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * (http://www.eclipse.org/legal/epl-v10.html), and the Refractions BSD
+ * License v1.0 (http://udig.refractions.net/files/bsd3-v10.html).
+ */
+package org.locationtech.udig.project.internal.render.impl.renderercreator;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.geotools.data.FeatureSource;
+import org.geotools.util.Range;
+import org.locationtech.udig.project.ILayer;
+import org.locationtech.udig.project.internal.render.Renderer;
+import org.locationtech.udig.project.render.AbstractRenderMetrics;
+import org.locationtech.udig.project.render.IRenderContext;
+import org.locationtech.udig.project.render.IRenderMetricsFactory;
+import org.locationtech.udig.project.render.IRenderer;
+
+/**
+ * For testing.  Creates a normal Renderer.  Accepts resources that resolve to FeatureSource objects.
+ * 
+ * @author Nikolaos Pringouris <nprigour@gmail.com>
+ */
+public class SingleFeatureRenderMetricsFactory2 implements IRenderMetricsFactory {
```

Comment:
Whats the different between SingleFeatureRenderMetricsFactory and SingleFeatureRenderMetricsFactory2?

Reply:
actually the contain an AbstractRenderMetrics with different values for certain metrics (latency & timeToDrawMetric) so that we are able to check explicit assignment via StyleBoardParameter and automatic via render metric

Reply:
Does it makes sense to rename these classes for better readability? Like Latency.. TimeToDrawRendereMetricsFactory